### PR TITLE
access dialogues via handler

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

Make dialogues accessible via their handler

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: aea/skills/base.py [hunk 1]
```diff
@@ -510,6 +510,14 @@ def parse_module(  # pylint: disable=arguments-differ
         """
         return _parse_module(path, handler_configs, skill_context, Handler)
 
+    @property
+    def protocol_dialogues(self) -> "Dialogues":  # type: ignore # noqa: F821
+        """Protocol dialogues"""
+        if self.SUPPORTED_PROTOCOL is None:
+            raise ValueError(f"SUPPORTED_PROTOCOL not set on {self}")
+        attribute = cast(PublicId, self.SUPPORTED_PROTOCOL).name + "_dialogues"
```
Comment (DavidMinarsch) [inline aea/skills/base.py:518]: On consideration, this doesn't make much sense. The attribute is user defined via the skill.yaml.

Comment (DavidMinarsch) [reply aea/skills/base.py:518]: Will keep open but won't merge for now as a result.

File: aea/skills/base.py [hunk 2]
```diff
@@ -510,6 +510,14 @@ def parse_module(  # pylint: disable=arguments-differ
         """
         return _parse_module(path, handler_configs, skill_context, Handler)
 
+    @property
+    def protocol_dialogues(self) -> "Dialogues":  # type: ignore # noqa: F821
```
Comment (DavidMinarsch) [inline aea/skills/base.py:514]:
```suggestion
    def protocol_dialogues(self, attribute: Optional[str] = None) -> "Dialogues":  # type: ignore # noqa: F821
```

File: aea/skills/base.py (hunk 1 above)
Comment (DavidMinarsch) [inline aea/skills/base.py:518]:
```suggestion
        attribute = cast(PublicId, self.SUPPORTED_PROTOCOL).name + "_dialogues" if attribute is None else attribute
```

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

File: aea/skills/base.py (hunk 2 above)
Comment (angrybayblade) [reply aea/skills/base.py:514]: Circular import issue

File: aea/skills/base.py [hunk 3]
```diff
@@ -510,6 +510,17 @@ def parse_module(  # pylint: disable=arguments-differ
         """
         return _parse_module(path, handler_configs, skill_context, Handler)
 
+    def protocol_dialogues(self, attribute: Optional[str] = None):  # type: ignore
```
Comment (angrybayblade) [inline aea/skills/base.py:513]: > type: ignore

Because of the circular import caused by `Dialogue` class cannot assign a return type to this

**Comment by DavidMinarsch** [comment]
Currently failing test: tests/test_skills/test_base.py::TestSkillProgrammatic::test_protocol_dialogues

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**